File: 20200821/Murph/Solution.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int MinDepth (TreeNode root) {
        if (root == null) return 0;
        var minTemplate = min (root);
        return minTemplate;
        int min (TreeNode node) {
            if (node.left == null && node.right == null) return 1;

            int minTT = int.MaxValue;
            if (node.left != null) minTT = Math.Min (min (node.left), minTT);

            if (node.right != null) minTT = Math.Min (min (node.right), minTT);

            return minTT + 1;
        }

    }
}

File: 20200821/Murph/Solution.go
func minDepth(root *TreeNode) int {
	if root == nil {
		return 0
	}
	return getMin(root)
}

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}

func getMin(node *TreeNode) int {

	if node.Left == nil && node.Right == nil {
		return 1
	}

	minTT := 99999

	if node.Left != nil {
		minTT = min(minTT, getMin(node.Left))
	}

	if node.Right != nil {
		minTT = min(minTT, getMin(node.Right))
	}

	return minTT + 1

}